core/src/main/java/org/apache/iceberg/MetadataUpdateParser.java

```diff
@@ -271,5 +378,51 @@ private static MetadataUpdate readAddSortOrder(JsonNode node) {
     UnboundSortOrder sortOrder = SortOrderParser.fromJson(sortOrderNode);
     return new MetadataUpdate.AddSortOrder(sortOrder);
   }
+
+  private static MetadataUpdate readSetDefaultSortOrder(JsonNode node) {
+    int sortOrderId = JsonUtil.getInt(SORT_ORDER_ID, node);
+    return new MetadataUpdate.SetDefaultSortOrder(sortOrderId);
+  }
+
+  private static MetadataUpdate readAddSnapshot(JsonNode node) {
+    Snapshot snapshot = SnapshotParser.fromJson(null, node);
```

Comment:
Looks like this is slightly wrong. The node passed in is the update, but it should pass `node.get(SNAPSHOT)` here instead.

Reply:
This seems to be working:

```
    Preconditions.checkArgument(node.has(SNAPSHOT), "Cannot parse missing field: snapshot");
    Snapshot snapshot = SnapshotParser.fromJson(null, node.get(SNAPSHOT));
```

Reply:
Updated as mentioned.